tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,237 @@
+    lr = LinearRegression()
+    with mlflow.start_run() as run:
+        lr_model = lr.fit(dataset_binomial)
+    run_id = run.info.run_id
+    run_data = get_run_data(run_id)
+    assert run_data.params == truncate_param_dict(
+        stringify_dict_values(_get_instance_param_map(lr))
+    )
+    assert run_data.tags == get_expected_class_tags(lr)
+    assert MODEL_DIR in run_data.artifacts
+    loaded_model = load_model_by_run_id(run_id)
+    assert loaded_model.stages[0].uid == lr_model.uid
+
+
+@pytest.mark.skipif(
+    LooseVersion(pyspark.__version__) < LooseVersion("3.1"),
+    reason="This test require spark version >= 3.1",
+)
+def test_models_in_allowlist_exist(spark_session):  # pylint: disable=unused-argument
+    mlflow.pyspark.ml.autolog()  # initialize the variable `mlflow.pyspark.ml._log_model_allowlist`
+
+    def model_does_not_exist(model_class):
+        module_name, class_name = model_class.rsplit(".", 1)
+        try:
+            module = importlib.import_module(module_name)
+            return not hasattr(module, class_name)
+        except ModuleNotFoundError:
+            return True
+
+    non_existent_classes = list(
+        filter(model_does_not_exist, mlflow.pyspark.ml._log_model_allowlist)
+    )
+    assert len(non_existent_classes) == 0, "{} in log_model_allowlist don't exist".format(
+        non_existent_classes
+    )
+
+
+def test_autolog_does_not_terminate_active_run(dataset_binomial):
+    mlflow.pyspark.ml.autolog()
+    mlflow.start_run()
+    lr = LinearRegression()
+    lr.fit(dataset_binomial)
+    assert mlflow.active_run() is not None
+    mlflow.end_run()
+
+
+def test_meta_estimator_fit_performs_logging_only_once(dataset_binomial):
+    from pyspark.ml.classification import LinearSVC, OneVsRest
+
+    mlflow.pyspark.ml.autolog()
+    with mock.patch("mlflow.log_params") as mock_log_params, mock.patch(
+        "mlflow.set_tags"
+    ) as mock_set_tags, mock.patch("mlflow.spark.log_model") as mock_log_model:
+        with mlflow.start_run() as run:
+            svc = LinearSVC()
+            ova = OneVsRest(classifier=svc)
+            ova.fit(dataset_binomial)
+
+        mock_log_params.assert_called_once()
+        mock_set_tags.assert_called_once()
+        mock_log_model.assert_called_once()
+
+        query = "tags.{} = '{}'".format(MLFLOW_PARENT_RUN_ID, run.info.run_id)
+        assert len(mlflow.search_runs([run.info.experiment_id])) == 1
+        assert len(mlflow.search_runs([run.info.experiment_id], query)) == 0
+
+
+def test_fit_with_params(dataset_binomial):
+    mlflow.pyspark.ml.autolog()
+    lr = LinearRegression()
+    extra_params = {lr.maxIter: 3, lr.standardization: False}
+    with mlflow.start_run() as run:
+        lr.fit(dataset_binomial, params=extra_params)
+    run_id = run.info.run_id
+    run_data = get_run_data(run_id)
+    assert run_data.params == truncate_param_dict(
+        stringify_dict_values(_get_instance_param_map(lr.copy(extra_params)))
+    )
```

Comment:
Just for completeness could we also assert that the model artifacts & tags are present here?